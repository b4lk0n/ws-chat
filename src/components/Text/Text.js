import React from 'react';
import cx from 'classnames';

import styles from './Text.module.css';

export const Text = ({ variant, className, children }) => (
  <span className={cx(styles.root, styles[variant], className)}>{children}</span>
);
