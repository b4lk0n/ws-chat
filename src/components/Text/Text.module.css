.root {
  display: inline-block;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans',
    'Droid Sans', 'Helvetica Neue', sans-serif;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  letter-spacing: 0.25px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.title {
  font-size: 16px;
  line-height: 24px;
  font-weight: 500;
}

.subtitle {
  font-size: 14px;
  line-height: 24px;
  font-weight: 500;
}

.caption {
  font-size: 12px;
  line-height: 16px;
}
