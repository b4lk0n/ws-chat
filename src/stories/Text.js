import React from 'react';
import { withKnobs, select, text } from '@storybook/addon-knobs';
import { Text } from '../components';

export default {
  title: 'Text',
  decorators: [withKnobs],
};

export const Default = () => (
  <Text
    variant={select('variant', {
      'â€”': undefined,
      title: 'title',
      subtitle: 'subtitle',
      caption: 'caption',
    })}
  >
    {text('children', 'Example text')}
  </Text>
);
