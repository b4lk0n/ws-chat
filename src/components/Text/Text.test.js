import React from 'react';
import { Text } from './Text';
import { render } from '@testing-library/react';

describe('<Text />', () => {
  it('should render the text', () => {
    const { container, getByText } = render(<Text>Hello</Text>);
    expect(getByText('Hello')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add additional class if provided', () => {
    const { container } = render(<Text className={'centered'}>Hello</Text>);
    expect(container.firstChild).toMatchSnapshot();
  });

  ['title', 'subtitle', 'caption'].forEach(variant => {
    it(`should render ${variant} text`, () => {
      const { container } = render(<Text variant={variant}>Hello</Text>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
