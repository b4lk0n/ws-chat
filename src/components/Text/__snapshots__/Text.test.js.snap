// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should add additional class if provided 1`] = `
<span
  class="root undefined centered"
>
  Hello
</span>
`;

exports[`<Text /> should render caption text 1`] = `
<span
  class="root caption"
>
  Hello
</span>
`;

exports[`<Text /> should render subtitle text 1`] = `
<span
  class="root subtitle"
>
  Hello
</span>
`;

exports[`<Text /> should render the text 1`] = `
<span
  class="root undefined"
>
  Hello
</span>
`;

exports[`<Text /> should render title text 1`] = `
<span
  class="root title"
>
  Hello
</span>
`;
